File: BackButtonHandler/BackButtonHandler/SecondViewController.m
//
//  SecondViewController.m
//  BackButtonHandler
//
//  Created by ZhiHua Shen on 2018/5/3.
//  Copyright © 2018年 ZhiHua Shen. All rights reserved.
//

#import "SecondViewController.h"
#import "UIViewController+BackButtonHandler.h"

@interface SecondViewController () <BackButtonHandlerProtocol>

@end

@implementation SecondViewController

- (void)viewDidLoad {
    [super viewDidLoad];

}

-(BOOL)navigationShouldPopOnBackButton {
    
    return YES;
}

@end

File: BackButtonHandler-Swift/BackButtonHandler-Swift/SecondViewController.swift
//
//  SecondViewController.swift
//  BackButtonHandler-Swift
//
//  Created by ZhiHua Shen on 2018/5/3.
//  Copyright © 2018年 ZhiHua Shen. All rights reserved.
//

import UIKit

class SecondViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()


    }

}

extension SecondViewController: BackButtonHandlerProtocol {
    
    func navigationShouldPopOnBackButton() -> Bool {
        return true
    }
    
}

